### PR TITLE
Most recent version of Favourite UC

### PR DESCRIPTION
**Pull request by izumidayo**

### CODE REVIEW

File: src/main/java/database/users.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+ userID,isAdmin,email,password,listPosts,listFavourites;
```
Comment (JohannLeidenfrost) [inline src/main/java/database/users.csv:1]: I think the semicolon at the end is unnecessary and may mess with reading the csv.

**Review by reganq — APPROVED** [review]
lgtm

**Review by reganq — APPROVED** [review]
Looks good to me!!

File: src/main/java/favourite/drivers/FavouriteDatabaseAccess.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            List<String[]> allUsers = readAllLines(Paths.get(partialPath+"users.csv"));
+            String[] userData = new String[6];
+            for (String[] user : allUsers) {
+                try {
+                    int id = Integer.parseInt(user[0]);
+                    if (id == userID) {
+                        userData = user;
+                    }
+                } catch (NumberFormatException ex) {
+                    System.out.println("Reading the header of the csv");
```
Comment (JohannLeidenfrost) [inline src/main/java/favourite/drivers/FavouriteDatabaseAccess.java:66]: You could skip this catch statement by skipping the first element in allUsers.

File: src/main/java/favourite/drivers/FavouriteDatabaseAccess.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            List<String[]> allPosts = readAllLines(Paths.get(partialPath+"posts.csv"));
+            String[] postData = new String[10];
+            for (String[] post : allPosts) {
+                try {
+                    int id = Integer.parseInt(post[0]);
+                    if (id == postID) {
+                        postData = post;
+                    }
+                } catch (NumberFormatException ex) {
+                    System.out.println("Reading the header of the csv");
```
Comment (JohannLeidenfrost) [inline src/main/java/favourite/drivers/FavouriteDatabaseAccess.java:99]: Same as above comment. I think it is a bit strange to print this every time you read the posts/user files.

File: src/test/java/favourite/FavouriteTest.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
```
Comment (JohannLeidenfrost) [inline src/test/java/favourite/FavouriteTest.java:19]: Can you add test coverage here?

File: src/main/java/delete_post/drivers/DeletePostDataAccess.java [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
-
-    }
-    public void removeTag(int postId, String tag){
-
-    }
-    public void deletePost(int postId){
-
-    }
-}
+//package delete_post.drivers;
```
Comment (sushimon) [inline src/main/java/delete_post/drivers/DeletePostDataAccess.java:1]: Why is everything commented out?

File: src/main/java/favourite/ui/FavouriteView.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    private final int postid;
+
+    /**
+     * Initializes FavouriteView
+     *
+     * @param postid id of the post being favourited/unfavourited
+     */
+    public FavouriteView(int postid, FavouriteController controller){
+        this.controller = controller;
+        this.postid = postid;
```
Comment (sushimon) [inline src/main/java/favourite/ui/FavouriteView.java:27]: Not in camel case

File: src/main/java/favourite/use_case/UserFactory.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        // converting string data into acceptable types to reconstruct a User object
+        boolean admin = Boolean.parseBoolean(userData[1]);
+        int id = Integer.parseInt(userData[0]);
+        String email = userData[2];
+        String password = userData[3];
+
+        // creating a List of Integers of ids of the user's posts from the datastring
+        String[] postids = userData[4].split(" ");
+        List<Integer> posts = new ArrayList<>();
+        for (String ids: postids){
```
Comment (sushimon) [inline src/main/java/favourite/use_case/UserFactory.java:28]: Not in camel case

**Review by sushimon — APPROVED** [review]
No glaring issues but check below for comments.

General comments/concerns:
- Did you accidentally comment out everything in `DeletePostDataAccess`?
- Some variables and methods are not declared in camel case.

**Pull request merged**